appformer-provisioning/appformer-provisioning-client/src/main/java/org/kie/appformer/provisioning/client/deployment/DeploymentPopupViewImpl.java

```diff
@@ -0,0 +1,298 @@
+    }
+
+    @Override
+    public String getHost( ) {
+        return hostTextInput.getValue();
+    }
+
+    @Override
+    public void setHost( String host ) {
+        hostTextInput.setValue( host );
+    }
+
+    @Override
+    public String getPort( ) {
+        return portTextInput.getValue();
+    }
+
+    @Override
+    public void setPort( String port ) {
+        this.portTextInput.setValue( port );
+    }
+
+    @Override
+    public String getManagementPort( ) {
+        return managementPortTextInput.getValue();
+    }
+
+    @Override
+    public void setManagementPort( String managementPort ) {
+        this.managementPortTextInput.setValue( managementPort );
+    }
+
+    @Override
+    public String getManagementUser( ) {
+        return managementUserTextInput.getValue();
+    }
+
+    @Override
+    public void setManagementUser( String managementUser ) {
+        this.managementUserTextInput.setValue( managementUser );
+    }
+
+    @Override
+    public String getManagementPassword( ) {
+        return managementPasswordTextInput.getValue();
+    }
+
+    @Override
+    public void setManagementPassword( String managementPassword ) {
+        this.managementPasswordTextInput.setValue( managementPassword );
+    }
+
+    @Override
+    public void loadDataSourceOptions( final List<Pair<String, String>> dataSourceOptions, final boolean addEmptyOption ) {
+        dataSourceSelector.clear();
+        if ( addEmptyOption ) {
+            dataSourceSelector.add( newOption( "", "" ) );
+        }
+        for ( Pair<String, String> optionPair: dataSourceOptions ) {
+            dataSourceSelector.add( newOption( optionPair.getK1(), optionPair.getK2() ));
+        }
+        refreshDataSourceSelector();
+    }
+
+    @Override
+    public String getDataSource( ) {
+        return dataSourceSelector.getValue();
+    }
+
+    @Override
+    public void setDataSource( String dataSource ) {
+        dataSourceSelector.setValue( dataSource );
+        refreshDataSourceSelector();
+    }
+
+    private Option newOption( final String text, final String value ) {
+        final Option option = new Option();
+        option.setValue( value );
+        option.setText( text );
+        return option;
+    }
+
+    private void refreshDataSourceSelector() {
+        Scheduler.get().scheduleDeferred( new Scheduler.ScheduledCommand() {
+            @Override
+            public void execute() {
+                dataSourceSelector.refresh();
+            }
+        } );
+    }
+
+    @Override
+    public void show() {
+        modal.show();
+    }
+
+    @Override
+    public void hide() {
+        modal.hide();
+    }
+
+    @Override
+    public void showMessage( String title, String message ) {
+        YesNoCancelPopup.newYesNoCancelPopup( title, message, null, null, ( org.uberfire.mvp.Command ) ( ) -> {
+            //do nothing.
+        } ).show();
+    }
+
+    @EventHandler( "local-server-radio-input" )
```

Comment:
I have to admit I did not think that using a `Widget` event handler would work on an element `@DataField`. That being said, if you are already using elements, you should use element event handlers (definitely is more future-proof). You could make this an element handler like so:

    @EventHandler( "local-server-radio-input" )
    private void onLocalServerRadioChange( @ForEvent( "change" ) org.jboss.errai.common.client.dom.Event event ) {
         presenter.onLocalChange();
    }